src/lib.rs

```diff
@@ -8,19 +10,58 @@ extern crate "unsafe-any" as uany;
 use std::any::Any;
 use std::intrinsics::TypeId;
 use std::collections::{hash_map, HashMap};
+use std::hash::{Hash, Hasher, Writer};
+use std::mem::transmute;
 
 // These traits are faster when we know the type is correct already.
 use uany::{UncheckedAnyDowncast, UncheckedAnyMutDowncast, UncheckedBoxAnyDowncast};
 
 use Entry::{Occupied, Vacant};
 
+/// Custom Hasher that does noop on TypeId
+///
+/// Copied from https://github.com/chris-morgan/anymap/blob/master/src/lib.rs
+#[deriving(Default,Copy)]
+pub struct TypeIdHasher;
+
+/// State for Custom Hasher
+///
+/// Copied from https://github.com/chris-morgan/anymap/blob/master/src/lib.rs
+#[deriving(Copy)]
+pub struct TypeIdState {
+    value: u64,
```

Comment:
This seems like it would be simpler if we just had `[u8; 8]` and could just write into it safely, then later just transmute to a u64.